### PR TITLE
Use DevExtreme EXXX messages instead of js exceptions for Text Editor buttons (T730671)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/ui/widget/ui.errors.js [hunk 1]
```diff
@@ -218,6 +218,46 @@ module.exports = errorUtils(errors.ERROR_MESSAGES, {
...
+
+    /**
+    * @name ErrorsUIWidgets.E1056
+    */
+    E1056: "Text Editor button's 'name' field must be a string",
+
+    /**
+    * @name ErrorsUIWidgets.E1057
+    */
+    E1057: "The 'buttons' item must have unique name, but the \"{0}\" value has dublicates",
```
Comment (DokaRus) [inline js/ui/widget/ui.errors.js]: du**p**licates?

**Review by DokaRus — COMMENTED** [review]
I think @RomanTsukanov should check errors description

File: js/ui/widget/ui.errors.js [hunk 2]
```diff
@@ -218,6 +218,46 @@ module.exports = errorUtils(errors.ERROR_MESSAGES, {
     */
     E1052: '{0} should have the "dataSource" option specified',
 
+    /**
+    * @name ErrorsUIWidgets.E1053
+    */
+    E1053: "The 'buttons' option must include an object or a string items only",
```
Comment (RomanTsukanov) [inline js/ui/widget/ui.errors.js]:
```suggestion
    E1053: "The \"buttons\" option accepts an array that contain only objects or string values",
```

File: js/ui/widget/ui.errors.js [hunk 3]
```diff
@@ -218,6 +218,46 @@ module.exports = errorUtils(errors.ERROR_MESSAGES, {
...
+
+    /**
+    * @name ErrorsUIWidgets.E1054
+    */
+    E1054: "Text Editor button's 'location' property can be 'after' or 'before' only",
```
Comment (RomanTsukanov) [inline js/ui/widget/ui.errors.js]: I don't think that we need this error. `location` has a default value. Use it whenever an invalid value is passed.

File: js/ui/widget/ui.errors.js [hunk 4]
```diff
@@ -218,6 +218,46 @@ module.exports = errorUtils(errors.ERROR_MESSAGES, {
...
+
+    /**
+    * @name ErrorsUIWidgets.E1056
+    */
+    E1056: "Text Editor button's 'name' field must be a string",
+
+    /**
+    * @name ErrorsUIWidgets.E1057
+    */
+    E1057: "The 'buttons' item must have unique name, but the \"{0}\" value has duplicates",
```
Comment (RomanTsukanov) [inline js/ui/widget/ui.errors.js]: Merge E1056 and E1057 into one error with the following message:

> One or several text editor buttons have invalid or non-unique \"name\" values.

File: js/ui/widget/ui.errors.js [hunk 5]
```diff
@@ -218,6 +218,46 @@ module.exports = errorUtils(errors.ERROR_MESSAGES, {
...
+
+    /**
+    * @name ErrorsUIWidgets.E1054
+    */
+    E1054: "Text Editor button's 'location' property can be 'after' or 'before' only",
+
+    /**
+    * @name ErrorsUIWidgets.E1055
+    */
+    E1055: "Text Editor button must have a name",
```
Comment (RomanTsukanov) [inline js/ui/widget/ui.errors.js]:
```suggestion
    E1055: "All text editor buttons must have names",
```

File: js/ui/widget/ui.errors.js [hunk 6]
```diff
@@ -218,6 +218,46 @@ module.exports = errorUtils(errors.ERROR_MESSAGES, {
...
+
+    /**
+    * @name ErrorsUIWidgets.E1057
+    */
+    E1057: "The 'buttons' item must have unique name, but the \"{0}\" value has duplicates",
+
+    /**
+    * @name ErrorsUIWidgets.E1058
+    */
+    E1058: "The predefined '{0}' button must have no 'location' or 'options' fields in the configuration",
```
Comment (RomanTsukanov) [inline js/ui/widget/ui.errors.js]: Why an error? Just don't apply `location` and `options`.

File: js/ui/widget/ui.errors.js [hunk 7]
```diff
@@ -218,6 +218,46 @@ module.exports = errorUtils(errors.ERROR_MESSAGES, {
...
+
+    /**
+    * @name ErrorsUIWidgets.E1059
+    */
+    E1059: "Editor does not support '{0}' action button",
+
+    /**
+    * @name ErrorsUIWidgets.E1060
+    */
+    E1060: "The 'buttons' option must be an array",
```
Comment (RomanTsukanov) [inline js/ui/widget/ui.errors.js]: Use E1053 instead

File: js/ui/widget/ui.errors.js [hunk 8]
```diff
@@ -218,6 +218,46 @@ module.exports = errorUtils(errors.ERROR_MESSAGES, {
...
+
+    /**
+    * @name ErrorsUIWidgets.E1058
+    */
+    E1058: "The predefined '{0}' button must have no 'location' or 'options' fields in the configuration",
+
+    /**
+    * @name ErrorsUIWidgets.E1059
+    */
+    E1059: "Editor does not support '{0}' action button",
```
Comment (RomanTsukanov) [inline js/ui/widget/ui.errors.js]: What editor?

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: js/ui/widget/ui.errors.js (hunk 8 above)
Comment (alexander-kotov-dx) [reply js/ui/widget/ui.errors.js]: This error should be generated if user want to use "spins" button for SelectBox for example.
Maybe "The "{0}" button type is not supported for this type of editor."

**Review by RomanTsukanov — DISMISSED** [review]

**Review by DokaRus — DISMISSED** [review]

**Review by KsyushaM — APPROVED** [review]

**Pull request merged**